pkg/upgrade/upgrade.go

```diff
@@ -615,3 +615,179 @@ func cleanupModelControllerLegacyDeployment(ctx context.Context, cli client.Clie
 
 	return nil
 }
+
+// TODO: to be removed: https://issues.redhat.com/browse/RHOAIENG-21080
+func PatchOdhDashboardConfig(ctx context.Context, cli client.Client) error {
+	log := logf.FromContext(ctx)
+
+	var dashboardConfigs unstructured.UnstructuredList
+	dashboardConfigs.SetGroupVersionKind(gvk.OdhDashboardConfig)
+
+	if err := cli.List(ctx, &dashboardConfigs); err != nil {
+		log.Error(err, "Failed to list odhdashboardconfig CRs")
+		return fmt.Errorf("error listing odhdashboardconfig CRs: %w", err)
+	}
+
+	for _, cr := range dashboardConfigs.Items {
+		if !cluster.IsNotReservedNamespace(&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: cr.GetNamespace()}}) {
```

Comment:
is this required ? my understanding is that there is only one `OdhDashboardConfig` per cluster so I thinnk it should be patched, regardless of the namespace ?

Reply:
I have changed the logic to handle the singleton case